zerver/actions/message_send.py

```diff
@@ -1397,6 +1409,13 @@ def check_send_message(
     skip_stream_access_check: bool = False,
     read_by_sender: bool = False,
 ) -> SentMessageResult:
+    if (
+        local_id
+        and sender_queue_id
+        and not message_is_idempotent(sender, local_id, sender_queue_id)
```

Comment:
`message_is_idempotent` is a weird name for this function; something like `was_request_already_sent` would be more appropriate.